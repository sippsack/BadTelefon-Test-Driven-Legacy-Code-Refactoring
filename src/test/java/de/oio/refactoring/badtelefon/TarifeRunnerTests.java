package de.oio.refactoring.badtelefon;

import static org.junit.Assert.assertEquals;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

import org.approvaltests.Approvals;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

public class TarifeRunnerTests {

	private PrintStream originalSysOut;
	private ByteArrayOutputStream consoleStream;

	@Before
	public void init() {
		originalSysOut = System.out;
		consoleStream = new ByteArrayOutputStream();
		PrintStream printStream = new PrintStream(consoleStream);
		System.setOut(printStream);
	}

	@Test
	public void testSimpleOutput() {
		System.out.println("Hallo Publikum!");
		System.out.print("Hallo Falk!");
		assertEquals("Hallo Publikum!\r\nHallo Falk!", consoleStream.toString());
	}

	@Test
	public void testRunnerWithApprovals() throws Exception {
		TarifeRunner.main(null);
		Approvals.verify(consoleStream.toString());
	}

	@After
	public void teardown() {
		System.setOut(originalSysOut);
	}

}
