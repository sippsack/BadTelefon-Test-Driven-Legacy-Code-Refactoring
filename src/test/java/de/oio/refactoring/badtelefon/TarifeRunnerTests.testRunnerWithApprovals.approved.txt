
Verarbeitung von Tarif 0
Berechne Gespräch mit 10 min um 12:30 mit Tarif 0
Preis für das Gespräch: 17,91
Gesamtgebühr nach Gespräch um 12:30 (Mondscheinzeit: false): 17,91
Berechne Gespräch mit 01 min um 09:45 mit Tarif 0
Preis für das Gespräch: 0,00
Gesamtgebühr nach Gespräch um 09:45 (Mondscheinzeit: false): 17,91
Berechne Gespräch mit 03 min um 18:00 mit Tarif 0
Preis für das Gespräch: 3,98
Gesamtgebühr nach Gespräch um 18:00 (Mondscheinzeit: false): 21,89
Berechne Gespräch mit 02 min um 17:45 mit Tarif 0
Preis für das Gespräch: 1,99
Gesamtgebühr nach Gespräch um 17:45 (Mondscheinzeit: false): 23,88
Berechne Gespräch mit 06 min um 23:30 mit Tarif 0
Preis für das Gespräch: 3,45
Gesamtgebühr nach Gespräch um 23:30 (Mondscheinzeit: true): 27,33
Abrechnung: 27.33

Verarbeitung von Tarif 1
Berechne Gespräch mit 10 min um 12:30 mit Tarif 1
Preis für das Gespräch: 12,90
Gesamtgebühr nach Gespräch um 12:30 (Mondscheinzeit: false): 12,90
Berechne Gespräch mit 01 min um 09:45 mit Tarif 1
Preis für das Gespräch: 1,29
Gesamtgebühr nach Gespräch um 09:45 (Mondscheinzeit: false): 14,19
Berechne Gespräch mit 03 min um 18:00 mit Tarif 1
Preis für das Gespräch: 3,87
Gesamtgebühr nach Gespräch um 18:00 (Mondscheinzeit: false): 18,06
Berechne Gespräch mit 02 min um 17:45 mit Tarif 1
Preis für das Gespräch: 2,58
Gesamtgebühr nach Gespräch um 17:45 (Mondscheinzeit: false): 20,64
Berechne Gespräch mit 06 min um 23:30 mit Tarif 1
Preis für das Gespräch: 4,74
Gesamtgebühr nach Gespräch um 23:30 (Mondscheinzeit: true): 25,38
Abrechnung: 25.380000000000003

Verarbeitung von Tarif 2
Berechne Gespräch mit 10 min um 12:30 mit Tarif 2
Preis für das Gespräch: 6,90
Gesamtgebühr nach Gespräch um 12:30 (Mondscheinzeit: false): 6,90
Berechne Gespräch mit 01 min um 09:45 mit Tarif 2
Preis für das Gespräch: 0,69
Gesamtgebühr nach Gespräch um 09:45 (Mondscheinzeit: false): 7,59
Berechne Gespräch mit 03 min um 18:00 mit Tarif 2
Preis für das Gespräch: 2,07
Gesamtgebühr nach Gespräch um 18:00 (Mondscheinzeit: false): 9,66
Berechne Gespräch mit 02 min um 17:45 mit Tarif 2
Preis für das Gespräch: 1,38
Gesamtgebühr nach Gespräch um 17:45 (Mondscheinzeit: false): 11,04
Berechne Gespräch mit 06 min um 23:30 mit Tarif 2
Preis für das Gespräch: 4,14
Gesamtgebühr nach Gespräch um 23:30 (Mondscheinzeit: true): 15,18
Abrechnung: 15.18
